File: shadowed-config-registries/src/main/java/io/github/legacy_fabric_community/shadowedconfig/registries/MutableRegistry.java
package io.github.legacy_fabric_community.shadowedconfig.registries;

import net.minecraft.util.Identifier;

public interface MutableRegistry<T> extends IdentifiableRegistry<T> {
	<V extends T> V set(int rawId, Identifier id, V entry);

	<V extends T> V add(Identifier id, V entry);

	boolean isEmpty();
}

File: shadowed-config-registries/src/main/scala/io/github/legacy_fabric_community/shadowedconfig/registries/MutableRegistry.scala
package io.github.legacy_fabric_community.shadowedconfig.registries

import net.minecraft.util.Identifier

trait MutableRegistry[T] extends IdentifiableRegistry[T] {
	def set[V <: T](rawId: Int, id: Identifier, entry: V): V

	def add[V <: T](id: Identifier, entry: V): V

	def isEmpty: Boolean
}
